agents/aliyun/fence_aliyun.py

```diff
@@ -153,18 +165,23 @@ def main():
 
 	run_delay(options)
 
-	if "--region" in options:
-		region = options["--region"]
-		if "--access-key" in options and "--secret-key" in options:
-			access_key = options["--access-key"]
-			secret_key = options["--secret-key"]
-			conn = client.AcsClient(access_key, secret_key, region)
-		elif "--ram-role" in options:
-			ram_role = options["--ram-role"]
-			role = EcsRamRoleCredential(ram_role)
-			conn = client.AcsClient(region_id=region, credential=role)
-		region_provider.modify_point('Ecs', region, 'ecs.%s.aliyuncs.com' % region)
-		
+	if "--region" not in options or options["--region"] == "":
+		fail_usage("Failed: option '--region' is invalid")
+
+	region = options["--region"]
+	if "--access-key" in options and "--secret-key" in options:
+		access_key = options["--access-key"]
+		secret_key = options["--secret-key"]
+		conn = client.AcsClient(access_key, secret_key, region)
+	elif "--ram-role" in options:
+		ram_role = options["--ram-role"]
+		role = EcsRamRoleCredential(ram_role)
+		conn = client.AcsClient(region_id=region, credential=role)
+	else:
+		fail_usage("Failed: User credentials are not set. Please set the Access Key and the Secret Key, or configure the RAM role.")
+
+	region_provider.modify_point('Ecs', region, 'ecs.%s.aliyuncs.com' % region)
```

Comment:
This should be in an if, so we dont fail on outputting metadata during e.g. build or if some distro provides the agent but user has to install the libraries to use it.

Reply:
I added a try/except, please help review whether this is appropriate